server/src/publications/dashboardData.js

```diff
@@ -0,0 +1,48 @@
+import {AccountingEntries} from "../../../imports/api/accountingEntries";
+import {InstallmentsCollection} from "../../../imports/api/installments";
+import {Accounts} from "../../../imports/api/accounts";
+import {Categories} from "../../../imports/api/categories";
+
+Meteor.publish("dashboardData.fetchAll", function(filters) {
+    const { payed = false, received = false , startRange, endRange} = filters;
+    const dateQuery = {
+        dueDate: { $gte: startRange, $lte: endRange},
+    };
+    const creditEntries = AccountingEntries.find({
+        credit: true,
+        payed: received,
+        ...dateQuery,
+    }).fetch();
+    const debitEntries = AccountingEntries.find({
+        credit: false,
+        payed,
+        ...dateQuery,
+    }).fetch();
+
+    const installments = InstallmentsCollection.find().fetch();
+    const accounts = Accounts.find().fetch();
+    const categories = Categories.find().fetch();
+
+
+    creditEntries.forEach((currentVal) => {
```

Comment:
same thing here, you return the cursors. But here, even the query is wrong. Using meteor subs/pub is like an idempotent way to make queries. You publish a set of data, and then you have to query that data on the client. You don't have to create another collection for credit/debit entries, but you have to **publish the page month accounting entries** and then the client can filter which of those are credit, and debit.

Reply:
But would it not be better to serve the data already filtered by the server? This means less work for the client, right? Or is it a bad pratice to have client only collections?